pkg/auth/spire/delegate.go

```diff
@@ -168,12 +174,9 @@ func (s *SpireDelegateClient) listenForBundleUpdates(ctx context.Context, errorC
 func (s *SpireDelegateClient) handleX509SVIDUpdate(svids []*delegatedidentityv1.X509SVIDWithKey) {
 	newSvidStore := map[string]*delegatedidentityv1.X509SVIDWithKey{}
 
+	s.svidStoreMutex.RLock()
 	for _, svid := range svids {
 
-		s.log.Debugf("processing spiffe://%s%s, Expires at %s", svid.X509Svid.Id.TrustDomain,
-			svid.X509Svid.Id.Path,
-			time.Unix(svid.X509Svid.ExpiresAt, 0))
-
 		if svid.X509Svid.Id.TrustDomain != s.cfg.SpiffeTrustDomain {
 			s.log.Debugf("skipping X509-SVID update for trust domain %s as it does not match ours", svid.X509Svid.Id.TrustDomain)
 			return
```

Comment:
Kinda nervous to release the lock in between reading and update, but I guess we're handling svid updates sequentially enough so that's not a problem. Here we should release though
```suggestion
			s.svidStoreMutex.RUnlock()
			return
```